DatadogRUM/Sources/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -157,9 +157,19 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
         interactionToNextViewMetric?.trackViewStart(at: startTime, name: name, viewID: viewUUID)
 
         self.viewEndedMetric = dependencies.viewEndedMetricFactory()
-        self.viewHitchesMetric = dependencies.viewHitchesMetricFactory()
-
-        if let viewHitchesMetric { dependencies.renderLoopObserver?.register(viewHitchesMetric) }
+        self.viewHitchesReader = dependencies.viewHitchesReaderFactory()
+
+        if let viewHitchesReader {
+            self.viewEndedMetric.add(
+                metric: ViewHitchesMetric(
+                    maxCount: viewHitchesReader.config.maxCollectedHitches,
+                    slowFrameThreshold: viewHitchesReader.config.acceptableLatency.toInt64Nanoseconds,
+                    maxDuration: viewHitchesReader.config.hangThreshold.toInt64Nanoseconds,
+                    viewMinDuration: Constants.minimumTimeSpentForRates.toInt64Nanoseconds
+                )
+            )
+            dependencies.renderLoopObserver?.register(viewHitchesReader)
+        }
```

Comment:
suggestion/ This setup doesn't utilize any internal state from `RUMViewScope`, which suggests it may not need to reside there. To streamline the architecture, consider moving the registration of the hitches metric (`.add(...)`) to `RUMFeature`, similar to how the "view ended" metric is handled:

```swift
// in RUMFeature.swift
controller.add(metric: ViewEndedMetric(...))
controller.add(metric: ViewHitchesMetric(...))
```